lib/jekyll/filters/url_filters.rb

```diff
@@ -33,6 +33,22 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces a path relative to the current page. For example, if the path
+      # `/images/me.jpg` is supplied, then the result could be, `me.jpg`,
+      # `images/me.jpg` or `../../images/me.jpg` depending on whether the current
+      # page lives in `/images`, `/` or `/sub/dir/` respectively.
+      #
+      # input - a path relative to the project root
+      #
+      # Returns the supplied path relativized to the current page.
+      def relativize_url(input)
+        return if input.nil?
+        input = ensure_leading_slash(input)
+        page_url = @context.registers[:page]["url"]
+        page_dir = Pathname(page_url).parent
+        Pathname(input).relative_path_from(page_dir).to_s
+      end
```

Comment:
some of the assignments can be done away:
```ruby
      def relativize_url(input)
        return if input.nil?
        page_dir = Pathname(@context.registers[:page]["url"]).parent
        Pathname(ensure_leading_slash(input)).relative_path_from(page_dir).to_s
      end
```
But I'll leave it to you to decide if the above hampers readability..